resources/benchmark-runner.mjs

```diff
@@ -265,56 +311,52 @@ export class BenchmarkRunner {
 
     async _runTestAndRecordResults(suite, test) {
         /* eslint-disable-next-line no-async-promise-executor */
-        return new Promise(async (resolve) => {
-            if (this._client?.willRunTest)
-                await this._client.willRunTest(suite, test);
-
-            setTimeout(() => {
-                this._runTest(suite, test, this._page, resolve);
-            }, 0);
-        });
-    }
+        if (this._client?.willRunTest)
+            await this._client.willRunTest(suite, test);
 
-    // This function ought be as simple as possible. Don't even use Promise.
-    _runTest(suite, test, page, testDoneCallback) {
         // Prepare all mark labels outside the measuring loop.
         const startLabel = `${suite.name}.${test.name}-start`;
         const syncEndLabel = `${suite.name}.${test.name}-sync-end`;
         const asyncStartLabel = `${suite.name}.${test.name}-async-start`;
         const asyncEndLabel = `${suite.name}.${test.name}-async-end`;
 
-        performance.mark(startLabel);
-        const syncStartTime = performance.now();
-        test.run(page);
-        const syncEndTime = performance.now();
-        performance.mark(syncEndLabel);
-
-        const syncTime = syncEndTime - syncStartTime;
-
-        performance.mark(asyncStartLabel);
-        const asyncStartTime = performance.now();
-        setTimeout(() => {
+        let syncTime;
+        let asyncStartTime;
+        let asyncTime;
+        const invokerClass = params.measurementMethod === 'raf' ? RAFTestInvoker : TimerTestInvoker;
+        const invoker = new invokerClass(() => {
```

Comment:
As on the other PR, maybe named-methods + bind makes this more readable?

Reply:
What do you mean by that? Could you show me an example?

Reply:
Along these lines (implying moving the *time values onto the class):

```
   const invoker = new invokerClass(this.measureSync.bind(this), this.measureAsync.bind(this), this._report.bind(this));
    ...
  }
  measureSync(...) {
   ...
  }
  measureAsync(...) {
    ...
  }
  ...
```
Maybe easier: using named arrow functions would be an improvement:

```
   ...
   const syncStart = () => {...};
   const measureAsync = () => {...};
   const report = () => {...};
   const invoker = new invokerClass(measureSync, measureAsync, report);
   
```

Reply:
.bind() makes it harder to read ;) 
But that latter case looks better to me.

Reply:
Done that.